### PR TITLE
[CI] run pytest in parallel

### PR DESCRIPTION
**Pull request by szha**
## Description ##
run pytest in parallel

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [x] run small tests in parallel
- [x] mark tests that require more resources
- [x] run large tests in serial
- [x] use built-in tmpdir fixtures for temporary files

### CODE REVIEW

File: tests/python/train/test_resnet_aug.py [hunk 1]
```diff
@@ -39,14 +40,15 @@ def get_net():
     return softmax
 
 # check data
-get_cifar10()
+path = tempfile.mkdtemp()
```
Comment (leezu) [inline tests/python/train/test_resnet_aug.py:43]: You're not cleaning up these temporary directories. You could use pytests functionality which will handle cleanup https://docs.pytest.org/en/latest/tmpdir.html

Comment (szha) [reply tests/python/train/test_resnet_aug.py:43]: thanks. this is useful. I will replace other occurrences.

**Comment by szha** [comment]
will look into it in a follow-up PR.

**Comment by szha** [comment]
@PatricZhao I noticed that the MKL/MKLDNN tests are taking a lot longer than non-MKL builds in the parallel test setting. I will try to run a couple more times to verify so this is just FYI.

**Comment by szha** [comment]
Update: the MKLDNN builds are actually executing a different set of tests which could explain the time difference. However, the MKL build is indeed executing the same unittest as regular python 3 CPU build and it's consistently taking a lot longer.

**Comment by marcoabreu** [comment]
Could that be due to cold booz? Doesn't mkl generate kernels the first time
you invoke them?

Sheng Zha <notifications@github.com> schrieb am Di., 5. Mai 2020, 21:13:

**Comment by szha** [comment]
@marcoabreu indeed that could be a likely cause. I reported my findings in #18244 and we can continue the discussion there.

**Pull request merged**